lib/DidStrategy.js

```diff
@@ -55,37 +61,34 @@ Strategy.prototype.authenticate = function(req) {
   // check POST data for an identity with a DID and a valid signature
   // pass to credentials-io for verification of the identity and all
   // credentials
-  async.auto({
-    verify: callback => {
-      const identity = req.body;
-      cio.verify(identity, callback);
-    }
-  }, (err, results) => {
-    if(err) {
-      return self.error(err);
-    }
-    if(!results.verify.verified) {
-      return self.error(new BedrockError(
-        'Identity credential signature verification failed.',
-        'SignatureNotVerified'));
-    }
-    if(!domains.includes(results.verify.domain)) {
-      return self.error(new BedrockError(
-        'Identity credential signature domain mismatch.',
-        'DomainMismatch', {
-          givenDomain: results.verify.domain,
-          expectedDomain: domains
-        }));
-    }
-    // TODO: need to support use case of "upgrading" from being authenticated
-    //   as the owner of an `account` to ALSO being the owner of an `identity`
-    //   that is not yet managed by the `account` ... only then will all the
-    //   capabilities required be present to start managing an identity
-    // TODO: see `httpSignatureStrategy` to add `account` and `actor`
-    req.user = {
-      identity: results.verify.identity,
-      credentials: results.verify.credentials
-    };
-    self.success(req.user);
-  });
+  let results;
+  try {
+    results = await cioVerify(req.body);
+  } catch(e) {
+    return self.error(e);
+  }
+
+  if(!results.verify.verified) {
+    return self.error(new BedrockError(
+      'Identity credential signature verification failed.',
+      'SignatureNotVerified'));
```

Comment:
Since this is a major release, do you want to standardize this error?  (and maybe other errors in this module)